### ISSUE TITLE
Request new terms to describe machine learning

### ISSUE THREAD

**Issue by esperetta** [evidenceontology/evidenceontology#301]
We would like to requests some new terms to help users distinguish different machine learning annotations systems we are using in UniProt and GO. After our last meeting, we would like to follow your suggestion and create some new terms, as children of COMPUTATIONAL EVIDENCE (ECO:0007672). 

We have some open questions we would like to discuss, but this is a proposed list. 

NEW TERM PARENT: ARTIFICIAL INTELLIGENCE METHOD  

A type of computational evidence where data are produced, and/or generated, and/or analysed by an artificial intelligence system capable of intelligent behaviour similar to what human and animal minds can do.  Examples might include machine learning, rules engines, expert systems, knowledge graphs and symbolic artificial Intelligence.
Reference:
Goel AK, Davies J (2019) Artificial intelligence. In: The Cambridge Handbook of Intelligence. Cambridge University Press

NEW TERM PARENT : MACHINE LEARNING METHOD

A type of artificial intelligence which captures the application of a broad suite of machine learning models trained on a combination of existing data, such as proteins or nucleic acid sequences, and other existing information about an entity, and is applied to the generation of new labels, numerical or textual or categorical about this entity.

CHILD 1: ARTIFICIAL NEURAL NETWORKS METHOD

A type of machine learning method which leverages a broad suite of neural network models, which use a combination of data attributes of an entity called features as inputs, and map them to outputs through a collection of connected hidden units called artificial neurons, whose parameters are learned from data.

CHILD 1A: DEEP LEARNING  METHOD

A type of artificial neural network which uses a combination of data attributes as input layer, multi layers, and an output layer. Each layer of nodes trains on a distinct set of features based on the previous layer’s output. Requires large amounts of data. 

Reference:
https://www.uniprot.org/help/ProtNLM

Andreea Gane, Maxwell L. Bileschi, David Dohan, Elena Speretta, Amélie Héliou, Laetitia Meng-Papaxanthos, Hermann Zellner, Eugene Brevdo, Ankur Parikh, Maria J. Martin, Sandra Orchard, UniProt Collaborators and Lucy J. Colwell, ProtNLM: Model-based Natural Language Protein, Preprint (Link to add (https://storage.googleapis.com/brain-genomics-public/research/proteins/protnlm/uniprot_2022_04/protnlm_preprint_draft.pdf)

CHILD 2: Rule-based machine learning evidence

A type of machine learning which uses a combination of data information and creates a set of rules which collectively represent the prediction model and are in the form of “IF:THEN” statements. The knowledge applied can be traced back to the rules and the original data. 

Reference: 
https://borgelt.net/apriori.html
https://www.uniprot.org/help/arba

**Comment by mgiglio99** [comment]
Hi Elena,
Suvvi and I will take a look at these.
We look forward to meeting with you to discuss these soon.
Thanks,
Michelle

**Comment by mlbileschi** [comment]
Commenting to make sure I'm subscribed to this discussion.
Nice to meet you, Michelle. (I'm an author on the publication mentioned on the Google side of the collaboration)

**Comment by mgiglio99** [comment]
Hi @esperetta and @mlbileschi 
These terms look good to us.
We would add the word "evidence" to the end of each term name.
And there are a couple of tiny changes we might want to make to the definitions.
But otherwise I think these will work just fine.
We look forward to chatting tomorrow and hearing what other questions you have.

**Comment by esperetta** [comment]
@mgiglio99 Thank you again to you and Suvvi for your feedback. As discussed, we have modified the terms by adding "evidence" at the end of each proposed terms and tried to better explain the meaning of "intelligence" in the definition of Artificial Intelligence. We have also added some references. 
As you request, we are adding here the link to the google document for the public so that the proposed image of the tree is also available. Looking forward to your comments. Thank you very much.

**Comment by esperetta** [comment]
https://docs.google.com/document/d/1-U2DpRrJBkZSbdUl7NDsIDRDWjtFUxPzgyGKCTEymX4/edit?usp=sharing

**Comment by mgiglio99** [comment]
Hi Elena and team,
These are looking good. I still worry that two of the definitions are a bit circular in that definitions shouldn't really include the terms they are meant to define. To address that I suggest a couple of small changes - revised areas of text are in italics. I realize my suggestions might not be exactly right, but I hope they give you the sense of what I'm going for.

1. 'artificial intelligence method evidence'
Proposed revised def: "A type of computational evidence where data are produced, and/or generated, and/or analysed by _an intelligence system that is manifested by a machine and which exhibits intelligent behaviors, similar to what human and animal minds can do including perceiving_, synthesizing, and inferring information.  Examples might include machine learning, rules engines, expert systems, knowledge graphs, and symbolic artificial Intelligence.

2. 'machine learning method evidence'
Proposed revised def:  "A type of artificial intelligence method evidence _in which computational systems build models based on sample data, known as training data, in order to make predictions or decisions without being explicitly programmed to do so. Such systems are_ trained on a combination of existing data, such as proteins or nucleic acid sequences, and other existing information about an entity, and are applied to the generation of new labels, numerical or textual or categorical about this entity.
(I took phrasing for the edit from wikipedia:  https://en.wikipedia.org/wiki/Machine_learning)
Also, I'm not entirely sure I understand the last part of the definition - "...a combination of existing data, such as proteins or nucleic acid sequences, and other existing information about an entity, and is applied to the generation of new labels, numerical or textual or categorical about this entity."  Do you mean that an example of the data is a protein sequence and that the "other existing information" might be metadata about the sequence (e.g. function)? 
Also, this is the only definition that refers specifically to biological examples. We're thinking either we should add some non-biology examples or make it more generic such as: "...a combination of existing data and metadata, and are applied to the generation of new labels, numerical or textual or categorical about an entity."

Thanks,
Michelle

**Comment by esperetta** [comment]
@mgiglio99 Thank you very much for your suggestions. I am just collecting some feedback internally and will post here the updated terms soon.

@vanaukenk  That's great news! Will try to keep terms as much inclusive as possible so they can be applied to different tasks.

**Comment by esperetta** [comment]
Hello,

**Comment by mgiglio99** [comment]
Hi Elena,
I think this all looks great.
These terms will be added for our next ECO release.
Best,
Michelle

**Comment by nsuvarnaiari** [comment]
Hi @esperetta 

All terms (including manual and automatic assertion terms have been made and pushed in April, 2023 release. Please let me know if something is incorrect.
Thanks for contacting us with new terms. It was nice working with you all.

Thanks Suvvi and Michelle

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/snadendla/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/snadendla/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


ECO_id| ECO_term
--|--
ECO:0008000 | machine learning method evidence
ECO:0008001 | deep learning method evidence
ECO:0008002 | rule-based machine learning method evidence
ECO:0008006 | deep learning method evidence used in automatic assertion
ECO:0008005 | deep learning method evidence used in manual assertion
ECO:0008004 | machine learning method evidence used in automatic assertion
ECO:0008003 | machine learning method evidence used in manual assertion
ECO:0008008 | rule-based machine learning method evidence used in automatic assertion
ECO:0008007 | rule-based machine learning method evidence used in manual assertion



</body>

</html>
